5.machine_learning/3.assess_generalizability/scripts/5.vis_dose_generalizability.r

```diff
@@ -36,35 +36,47 @@ combined_probabilities_df$Metadata_dose <- factor(combined_probabilities_df$Meta
 height <- 5
 width <- 20
 options(repr.plot.width = width, repr.plot.height = height)
-ridge_plot_healthy <- (
-    ggplot(combined_probabilities_df, aes(
-        x = Healthy_probas, 
-        y = Metadata_heart_number, 
-        fill = Metadata_treatment, 
-        grouped = Metadata_treatment))
-    + geom_density_ridges(
-        aes(fill = Metadata_treatment), alpha = 0.7, scale = 3, rel_min_height = 0.01, bandwidth = 0.1
-    )
-    + geom_vline(xintercept = 1, linetype = "dashed", color = "black")
-    + scale_x_continuous(breaks = seq(0, 1, 0.5))
-    + facet_grid(model_type ~ Metadata_dose, scales = "free_y")
-    + labs(y = "Heart")
-    + labs()
-    + theme_bw()
-    + theme(legend.position = "none",
-      axis.text = element_text(size = 12),
-      axis.text.x = element_text(size = 12),
-      axis.title = element_text(size = 14),
-      strip.text = element_text(size = 14),
-      strip.background = element_rect(
-          colour = "black",
-          fill = "#fdfff4"
-      )
-    )
+
+# Define colors for the doses
+dose_colors <- c(
+  "0 uM" = brewer.pal(8, "Dark2")[7],  # Unique color for 0 uM
+  "0.005 uM" = brewer.pal(8, "Dark2")[4], # Same color for all other doses
+  "0.014 uM" = brewer.pal(8, "Dark2")[4],
+  "0.04 uM" = brewer.pal(8, "Dark2")[4],
+  "0.12 uM" = brewer.pal(8, "Dark2")[4],
+  "0.37 uM" = brewer.pal(8, "Dark2")[4],
+  "1.11 uM" = brewer.pal(8, "Dark2")[4],
+  "3.33 uM" = brewer.pal(8, "Dark2")[4],
+  "5 uM" = brewer.pal(8, "Dark2")[4],
+  "10 uM" = brewer.pal(8, "Dark2")[4]
```

Comment:
```suggestion
# Define colors for the doses
zero_dose_color <- brewer.pal(8, "Dark2")[7],  # Unique color for 0 uM
all_other_doses_color <- brewer.pal(8, "Dark2")[4], # Same color for all other doses

dose_colors <- c(
  "0 uM" = zero_dose_color,
  "0.005 uM" = all_other_doses_color,
  "0.014 uM" = all_other_doses_color,
  "0.04 uM" = all_other_doses_color,
  "0.12 uM" = all_other_doses_color,
  "0.37 uM" = all_other_doses_color,
  "1.11 uM" = all_other_doses_color,
  "3.33 uM" = all_other_doses_color,
  "5 uM" = all_other_doses_color,
  "10 uM" = all_other_doses_color
```

Reply:
In the event you need to change this, you only need to change it once (also it does not violate DRY)

Reply:
Ohhh I like this change! So much more simple and concise, added!